src/bosh-director/spec/unit/cloud/external_cpi_spec.rb

```diff
@@ -644,4 +644,20 @@ def self.it_raises_an_error_with_ok_to_retry(error_class)
   describe '#info' do
     it_calls_cpi_method(:info)
   end
+
+  context 'Fix for a dealock scenario when a cpi sub-process sends an inclomplete line (missing "\n") to STDOUT' do
```

Comment:
This test still passes for me when I use `readline` instead of `readline_nonblock`. I would have expected it to fail.

Reply:
Well, the test just hangs because the whole thing deadlocks.

Reply:
Hm, interesting, I tried this on my machine yesterday and it deadlocked straight away.

Reply:
How about a timeout? It should throw an exception after 20 seconds and fail the test.